$ NetBSD $

pragma GCC diagnostic only acts on some lines.
See http://gcc.gnu.org/bugzilla/show_bug.cgi?id=52116

--- session.c.orig	2013-11-25 10:40:09.000000000 +0000
+++ session.c	2013-11-25 10:42:06.000000000 +0000
@@ -18,7 +18,7 @@
 #include "gtk2-compat.h"
 #endif
 
-#if defined(__GNUC__) || defined(__clang__)
+#if defined(__clang__)
 #define DO_PRAGMA(x) _Pragma(#x)
 #else
 #define DO_PRAGMA(x)
